src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java

```diff
@@ -665,17 +673,27 @@ private StartWorkflowExecutionResponse startWorkflowExecution(
   @Override
   public GetWorkflowExecutionHistoryResponse GetWorkflowExecutionHistoryWithTimeout(
       GetWorkflowExecutionHistoryRequest request, Long timeoutInMillis) throws TException {
-    return measureRemoteCall(
+    Map<String, String> tags = ImmutableMap.of(MetricsTag.RequestType, "normal");
+    if (request.isWaitForNewEvent()){
+      tags = ImmutableMap.of(MetricsTag.RequestType, "long-poll");
+    }
+    return measureRemoteCallWithTags(
         ServiceMethod.GET_WORKFLOW_EXECUTION_HISTORY,
-        () -> getWorkflowExecutionHistory(request, timeoutInMillis));
+        () -> getWorkflowExecutionHistory(request, timeoutInMillis),
+        tags);
   }
 
   @Override
   public GetWorkflowExecutionHistoryResponse GetWorkflowExecutionHistory(
       GetWorkflowExecutionHistoryRequest request) throws TException {
-    return measureRemoteCall(
-        ServiceMethod.GET_WORKFLOW_EXECUTION_HISTORY,
-        () -> getWorkflowExecutionHistory(request, null));
+    Map<String, String> tags = ImmutableMap.of(MetricsTag.RequestType, "normal");
```

Comment:
I don't know uber's standards around ternary statements but I'd be tempted to do:
`Map<String, String> tags = ImmutableMap.of(MetricsTag.RequestType, (request.isWaitForNewEvent() ? "long-poll" : "normal"));`
That would avoid running `ImmutableMap.of` twice and whatever performance penalty that entails.